### DISCUSSION TITLE
How to add redstone status to a square

### DISCUSSION THREAD

**Discussion by alazeprt** [FabricMC/fabric-api#2661] [Mod Dev Support]
I want my square to change a picture when it detects a red stone signal. How can I add a redstone signal detection?

**Comment by warjort** [comment]
Look at how `DispenserBlock` uses `World.isReceivingRedstonePower(BlockPos)` in the `neighborUpdate()` callback.

**Reply by alazeprt** [reply]
After I wrote a block class(Compress the class into this zip file),
[block.zip](https://github.com/FabricMC/fabric/files/10022207/block.zip)
 an error occurred during compilation
![Error Info](https://user-images.githubusercontent.com/92018941/202196846-0be99e38-7e74-4fd4-9f8f-2dc942be1814.png)
How to solve this problem?
